### PR TITLE
feat(edit-content): Add Workflow Action button #26969

### LINKED ISSUE #26969: Edit Content: Add Workflow Action button to Edit Content
### Parent Issue

#25445

### User Story

As a content creator, I want to be able to advance my content to the next stage or workflow, so that I can efficiently move my content through the system.

### Acceptance Criteria

- [ ] The split buttons should be added to the content editing interface (Use PrimeNg).
- [ ] When a split button is clicked, it should display a dropdown with options to advance the content to the workflow.
- [ ] The dropdown options should be configurable and customizable based on the specific content types and workflows.
- [ ] Clicking an option in the dropdown should trigger the appropriate action to advance the content to the selected stage or workflow.
- [ ] The split buttons should be accessible, adhering to WCAG 2.1 guidelines and providing proper keyboard navigation and focus management.
- [ ] Proper error handling should be implemented, providing helpful messages if there are any issues with advancing the content to the selected stage or workflow.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Figma Design: https://www.figma.com/file/YMTdSO5gKf18rc1pCCv1jN/Edit-Contentlet?type=design&node-id=638-21305&mode=design&t=GwsaZP0A6wFhZBxR-4

Worflow actions - Split Buttons

### Assumptions & Initiation Needs

- To retrieve the available actions for the current content, we must call the endpoint: `/api/v1/workflow/contentlet/{inode}/actions?renderMode=editing.`
- We create a collection of actions for each `SEPARATOR` action.
- We need a split button for each collection of actions.

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

- [ ] Add split buttons to the content editing interface
- [ ] Implement the dropdown functionality for the split buttons
- [ ] Configure and customize the dropdown options based on content types and workflows
- [ ] Implement responsive design for the split buttons and dropdown
- [ ] Implement accessibility features and keyboard navigation
- [ ] Handle error scenarios and provide helpful messages

### PR DESCRIPTION
**Pull request by rjvelazco**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at c849188</samp>

This pull request implements and integrates a new feature of grouping and separating workflow actions in the edit content toolbar, based on the metadata of the actions. It refactors the `DotWorkflowActionsFireService` and the `DotWorkflowsActionsService` to use the Angular `HttpClient` and the `SpectatorHttp` library for testing. It also improves the styles and layout of the `DotEditContentFormComponent` and the `EditContentLayoutComponent`. It modifies the `DotCMSWorkflowAction` interface to include a `metadata` property, and the `DotRouterService` spec file to mock the `firstChild` property of the route. It adds the `DotEditContentToolbarComponent` and its template, styles and spec file. It updates the `edit-content.layout.component.html` file to use the new component.

## Related Issue
Fixes #26969

# Explanation of Changes

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at c849188</samp>

*  Refactor `DotWorkflowActionsFireService` and `DotWorkflowsActionsService` to use `HttpClient` instead of `CoreWebService` and `SpectatorHttp` instead of `HttpTestingController` ([link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL1-R4),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL35-R137),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL178-R176),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL196-R213),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL251-R228),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL263-R235),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-177a9ad469f42dfc834e7e894ef4c4aab422edabb95cf978a5161c60d2e527baL3-R7),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-177a9ad469f42dfc834e7e894ef4c4aab422edabb95cf978a5161c60d2e527baL30-R31),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-177a9ad469f42dfc834e7e894ef4c4aab422edabb95cf978a5161c60d2e527baL46-R51),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-177a9ad469f42dfc834e7e894ef4c4aab422edabb95cf978a5161c60d2e527baL63-R64),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-177a9ad469f42dfc834e7e894ef4c4aab422edabb95cf978a5161c60d2e527baL173-R175),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-a3d679549be6d39c50646cbaa2171aaf830fae03f52fba5d34565ecdfeac8214L1-R15),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-a3d679549be6d39c50646cbaa2171aaf830fae03f52fba5d34565ecdfeac8214L37-R33),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-c9275b56ee8e16002cd65d33eb32c304750cecfc9611c54601f48dbdcc126936L3-R7),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-c9275b56ee8e16002cd65d33eb32c304750cecfc9611c54601f48dbdcc126936L17-R18),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-c9275b56ee8e16002cd65d33eb32c304750cecfc9611c54601f48dbdcc126936L27-R30),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-c9275b56ee8e16002cd65d33eb32c304750cecfc9611c54601f48dbdcc126936L49-R47))
* Add `metadata` property to `DotCMSWorkflowAction` interface to store additional information about the workflow action ([link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-fdae9a63e765099fd0e81ebbc200dae6f7332745353737ca01b436d21d2cf3f4R26))
* Implement and integrate `DotEditContentToolbarComponent` to display and fire workflow actions grouped and separated by subtype in the edit content toolbar ([link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-1a3e657016880498977ce74202425946ae817683630dd29a8b673b2dd4672e5aL6-R25),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-1a3e657016880498977ce74202425946ae817683630dd29a8b673b2dd4672e5aR34-R37),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-dc47489729e9c1942a358ded815735eff66a4fd2fece28f95923f46f75f0d6c0R1-R12),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-81dcc4b0c282554aed057702051b4fd5dcc46275afc9a4d16417e1d6a6777aedR1-R7),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-9c106511c221965d2f1b3d3709446e61bfd9d0ddec1aecb3d98d074491cc7ae4R1-R22),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-7453213c303f4ec188b861d614fdef0dd85978983a5aefd90526f5d904e83eefR1-R78),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-2e7135ceb3505b828ceedd8ae2a823ab1ca29b2ff6e8a107aeb29cdf6181be25L1-R8),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-0684fbc2e74e0b8885211e28ce424e7fc14fc29c6d0c3b9d7dc4e50906c08ea9L5-R10),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-0684fbc2e74e0b8885211e28ce424e7fc14fc29c6d0c3b9d7dc4e50906c08ea9L13-R26),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-0726627227f6f78ea4a1a2c83bc1bff882c631afd112ab9cac9214c1a770aa40R15),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-0726627227f6f78ea4a1a2c83bc1bff882c631afd112ab9cac9214c1a770aa40R24),[link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-0726627227f6f78ea4a1a2c83bc1bff882c631afd112ab9cac9214c1a770aa40L28-R30))
* Add `firstChild` property to `ActivatedRouteMock` class to mock the child route of the current route in the `DotRouterService` spec file ([link](https://github.com/dotCMS/core/pull/27005/files?diff=unified&w=0#diff-e37050a644fcd1af4f032c7e3e735b4a6ba11b13488e7f9fd75be7e6784eb352R58-R62))

### Video
https://github.com/dotCMS/core/assets/72418962/8c60eef4-e6ff-4af7-ad3f-0401f7fb35f4

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                            label: action.name,
+                            command: () => this.actionFired.emit(action)
+                        });
+                    }
+
+                    return acc;
+                },
+                [[]]
+            )
+            .filter((group) => group.length);
```
[the remark this answers is not in the document]
Comment (zJaaal) [reply core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.ts:58]: Here you are iterating twice the array. But it's just a nice to have, it's not a big deal, I don't think this actions are too long

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.scss [hunk 2]
```diff
@@ -2,15 +2,24 @@
 
 :host {
     display: grid;
-    grid-template-columns: 1fr 16rem;
-    gap: $spacing-4;
-    padding: $spacing-4;
+    grid-template-areas:
+        "header header"
+        "body sidebar";
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.scss:7]: NIce!

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts [hunk 3]
```diff
@@ -1,95 +1,85 @@
...
-                  }
-              })
-          )
-        : this.dotEditContentService
-              .getContentTypeFormData(this.contentType)
-              .pipe(
-                  map(({ layout, fields }) => ({ layout, fields, contentType: this.contentType }))
-              );
+    vm$: Observable<EditContentPayload> = this.store.vm$.pipe(
+        skip(1),
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts:64]: Maybe you can avoid this skip by lazy loading the store.

File: core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            );
+        }
+    );
+
+    private getNewContent(contentTypeVar: string) {
+        return forkJoin({
+            contentType: this.dotEditContentService.getContentType(contentTypeVar),
+            actions: this.workflowActionService.getDefaultActions(contentTypeVar),
+            contentlet: of(null)
+        });
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.ts:142]: TIL! I thought it just worked with arrays

File: core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+                        DotRenderMode.EDITING
+                    ),
+                    contentlet: of(contentlet)
+                });
+            })
+        );
+    }
+
+    private updateURL(inode: string) {
+        this.location.replaceState(`/content/${inode}`); // Replace the URL with the new inode without reloading the page
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.ts:179]: Nice!

File: dotCMS/src/main/webapp/WEB-INF/messages/Language.properties [hunk 6]
```diff
@@ -1193,6 +1193,7 @@ dot.common.message.locking=Locking...
 dot.common.message.pageurl.copy.clipboard=Copy Page URL
 dot.common.message.saved=All changes Saved
 dot.common.message.saving=Saving...
+dot.common.message.success=Succes
```
Comment (zJaaal) [inline dotCMS/src/main/webapp/WEB-INF/messages/Language.properties:1196]: Here is a typo. `Success`* Also there is `Success=Success!` not sure if we want it without the exclamation mark

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    <div class="p-toolbar-group-right">
+        <ng-container *ngIf="groupedActions.length">
+            <ng-container *ngFor="let items of groupedActions; index as i">
+                <p-splitButton
+                    [styleClass]="i !== 0 ? 'p-button-outlined' : ''"
+                    [model]="items | slice : 1"
+                    [label]="items[0].label"
+                    (onClick)="items[0].command()"></p-splitButton>
+            </ng-container>
+        </ng-container>
```
[the remark this answers is not in the document]
Comment (KevinDavilaDotCMS) [reply core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.html:12]: Why dont use ngfor on splitButton?

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    private groupActions(actions: DotCMSWorkflowAction[]): MenuItem[][] {
+        return actions
+            .reduce(
+                (acc, action) => {
+                    if (action?.metadata?.subtype === DotCMSActionSubtype.SEPARATOR) {
+                        acc.push([]);
+                    } else {
+                        acc[acc.length - 1].push({
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.ts:48]: The `.at(-1)` method is available from ECMAScript 2021 and we are using `es2015` @zJaaal

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.html:12]: I'll do it

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<p-toolbar class="header">
+    <div class="p-toolbar-group-left"></div>
+    <div class="p-toolbar-group-right">
+        <ng-container *ngIf="groupedActions.length">
+            <ng-container *ngFor="let items of groupedActions; index as i">
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.html:5]: Why dont use ngfor on splitButton?

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html [hunk 10]
```diff
@@ -1,15 +1,15 @@
...
+    <dot-edit-content-form class="body" [formData]="vm" (changeValue)="setFormValue($event)" />
+    <aside class="sidebar">
+        <dot-edit-content-aside [contentLet]="vm.contentlet" [contentType]="vm.contentType" />
     </aside>
-    <p *ngIf="isContentSaved">Content saved!</p>
 </ng-container>
 
 <ng-template #noContent>No content to show</ng-template>
+
+<p-toast></p-toast>
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html:15]: Don't we have this global in the whole app?

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.scss (hunk 2 above)
Comment (fmontes) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.scss:7]: Nice!

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts (hunk 3 above)
Comment (fmontes) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts:64]: Hujummm..

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts [hunk 11]
```diff
@@ -1,95 +1,85 @@
...
+            this.contentType = contentlet?.contentType || contentType?.variable;
+            this.inode = contentlet?.inode;
+        }),
+        map(({ actions, contentType, contentlet }) => ({
+            actions,
+            contentType: this.contentType,
+            layout: contentType?.layout || [],
+            fields: contentType?.fields || [],
+            contentlet
+        }))
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts:75]: Should all this be a selector?

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts [hunk 12]
```diff
@@ -101,4 +91,19 @@ export class EditContentLayoutComponent implements OnInit {
     setFormValue(formValue: Record<string, string>) {
         this.formValue = formValue;
     }
+
+    fireAction(action: DotCMSWorkflowAction): void {
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts:95]: doc.

File: core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.spec.ts [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+
+    it('should have a initial value', (done) => {
+        spectator.service.vm$.subscribe((state) => {
+            expect(state).toEqual({
+                actions: [],
+                contentType: null,
+                contentlet: null
+            });
+            done();
+        });
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.spec.ts:78]: lazy load better.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-custom-field/dot-edit-content-custom-field.component.ts [hunk 14]
```diff
@@ -37,14 +35,13 @@ import { DotEditContentService } from '../../services/dot-edit-content.service';
...
     @Input() field!: DotCMSContentTypeField;
+    @Input() contentType!: string;
 
     @ViewChild('iframe') iframe!: ElementRef<HTMLIFrameElement>;
 
     private controlContainer = inject(ControlContainer);
-    private editContentService = inject(DotEditContentService);
     private zone = inject(NgZone);
 
-    private contentType = this.editContentService.currentContentType;
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-custom-field/dot-edit-content-custom-field.component.ts:47]: Nice!

File: core-web/libs/edit-content/src/lib/models/dot-edit-content-form.interface.ts [hunk 15]
```diff
@@ -1,12 +1,14 @@
...
+    DotCMSContentlet,
+    DotCMSWorkflowAction
 } from '@dotcms/dotcms-models';
 
-export interface EditContentFormData {
+export interface EditContentPayload {
     layout: DotCMSContentTypeLayoutRow[];
     fields: DotCMSContentTypeField[];
     contentlet?: DotCMSContentlet;
+    actions?: DotCMSWorkflowAction[];
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/models/dot-edit-content-form.interface.ts:12]: Why optional? Don't we need this all the time?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/edit-content/src/lib/models/dot-edit-content-form.interface.ts (hunk 15 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/models/dot-edit-content-form.interface.ts:12]: Yes, I marked it as optional while I moving things around/refactoring

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.spec.ts [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+            imports: [DotEditContentToolbarComponent]
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(DotEditContentToolbarComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.spec.ts:20]: Right, I'll test the component

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html (hunk 10 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html:15]: We have one in the `main-legacy.component.ts`, do you want me to move it tho the main template?

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.spec.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/components/dot-edit-content-toolbar/dot-edit-content-toolbar.component.spec.ts]: Done!

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts:64]: Done!

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts (hunk 11 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts:75]: Done!

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts (hunk 12 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts:95]: Done!

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts [hunk 17]
```diff
@@ -1,95 +1,85 @@
...
-        : this.dotEditContentService
-              .getContentTypeFormData(this.contentType)
-              .pipe(
-                  map(({ layout, fields }) => ({ layout, fields, contentType: this.contentType }))
-              );
+    vm$: Observable<EditContentPayload> = this.store.vm$.pipe(
+        skip(1),
+        tap(({ contentType, contentlet }) => {
+            this.contentType = contentlet?.contentType || contentType?.variable;
+            this.inode = contentlet?.inode;
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.ts]: Done!

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-field/dot-edit-content-field.component.html [hunk 18]
```diff
@@ -63,6 +63,7 @@
     <dot-edit-content-custom-field
         *ngSwitchCase="fieldTypes.CUSTOM_FIELD"
         [field]="field"
+        [contentType]="contentType"
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/components/dot-edit-content-field/dot-edit-content-field.component.html:66]: Yeap!

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html [hunk 19]
```diff
@@ -2,8 +2,14 @@
...
-        (actionFired)="fireWorkflowAction($event)"></dot-edit-content-toolbar>
-
+        (actionFired)="
+            fireWorkflowAction({
+                actionId: $event.id,
+                inode: vm.contentlet.inode,
+                contentType: vm.contentType
+            })
+        "></dot-edit-content-toolbar>
+    <!--  "fireWorkflowAction($data, vm.inode, vm.contentType)-->
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html:12]: remove

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html (hunk 19 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.html]: Done!

**Review by fmontes — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**